.github/workflows/test-build.yml

```diff
@@ -0,0 +1,93 @@
+name: Test Build
+
+on:
+  workflow_dispatch:
+  pull_request:
+  push:
+
+jobs:
+  run-tests:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+        with:
+          ref: ${{ github.ref }}
+
+      - name: Setup Go
+        uses: actions/setup-go@v4
+        with:
+          go-version: 1.21.x
+
+      - name: Run linting tests
+        id: update
+        run: |
+          make prepare
+          make lint
+          make fmt-check
+          make imports-check
+
+      - name: Run unit tests
+        run: |
+          make coverage-ci
+
+      - name: Build cross-platform binaries
+        run: |
+          make build-cli-cross-platform
+
+      - name: Run integration tests
+        env:
+          CI_ACCOUNT: ${{ secrets.CI_ACCOUNT }}
+          CI_SUBACCOUNT: ${{ secrets.CI_SUBACCOUNT }}
+          CI_API_KEY: ${{ secrets.CI_API_KEY }}
+          CI_API_SECRET: ${{ secrets.CI_API_SECRET }}
+          LW_INT_TEST_AWS_ACC: ${{ secrets.LW_INT_TEST_AWS_ACC }}
+        run: |
+          sudo apt-get update
+          sudo apt-get install -y vim
+          make integration-only
+
+      # TODO Display test report
+
+  trigger-release:
+    runs-on: ubuntu-latest
+    container:
+      image: techallylw/ally-releases:1.21.0
+    steps:
+      - name: Trigger release
+        if: github.ref == 'refs/heads/main'
+        env:
+          GPG_SECRET_KEY: ${{ secrets.GPG_SECRET_KEY }}
+          GPG_SIGNING_KEY: ${{ secrets.GPG_SIGNING_KEY }}
+        run: |
+          echo "$GPG_SECRET_KEY" | base64 --decode | gpg --import --no-tty --batch --yes
+          scripts/release.sh trigger
+
+  notify-on-failure:
+    runs-on: ubuntu-latest
+    needs: [run-tests, trigger-release]
+    steps:
+      - name: Notify Slack on Failure
+        uses: slackapi/slack-github-action@v1.25.0
```

Comment:
Release/DXP would prefer that we use https://github.com/lacework-dev/actions/tree/main/report-to-slack-v1

That was the feedback I received in a PR for Github workflow changes

Reply:
uh.. is that recommended or mandatory? The lw report-to-slack action can't pick up the @ in the text. And fields are laid out in two columns in a wired order. 

<img width="707" alt="Screenshot 2024-02-14 at 12 04 14 PM" src="https://github.com/lacework/go-sdk/assets/7945666/95ae57d9-9144-44a7-8516-7e9cca2d0838">